src/web/resources/report.js

```diff
@@ -35,6 +35,94 @@ function Element(tagname, props, children) {
     return $elt;
 }
 
+var renames = {};
+renames["imp-start"] = "Improved start"
+renames["apx-start"] = "Approximate start"
+renames["uni-start"] = "Regressed from start"
+renames["ex-start"]  = "Exact start"
+renames["eq-start"]  = "Equal start"
+renames["lt-start"]  = "Less than start"
+renames["gt-start"]  = "Greater than start"
+renames["gt-target"] = "Greater than target"
+renames["eq-target"] = "Equal than target"
+renames["lt-target"] = "Less than target"
+renames["error"]     = "Error"
+renames["timeout"]   = "Timeout"
+renames["crash"]     = "Crash"
+
+const Filters = new Component("#filters", {
+    setup: function () {
+        // Warning fragile starting state
+        const regressedTags = ["uni-start", "lt-target", "lt-start", 
+        "apx-start", "timeout", "crash", "error"]
+        const improvedTags = ["imp-start", "ex-start", "eq-start", "eq-target",
+        "gt-target"]
+        const advancedTags = []
+        advancedTags.push(regressedTags)
+        advancedTags.push(improvedTags)
+
+        // build labels
+        const regressedLabel = this.buildCheckboxLabel("regressed", "Regressed", true)
+        const improvedLabel = this.buildCheckboxLabel("improved", "Improved", true)
+
+        // add listeners
+        regressedLabel.addEventListener("click", this.attachLeaderToChildren("regressed", regressedTags))
+        improvedLabel.addEventListener("click", this.attachLeaderToChildren("improved", improvedTags))
```

Comment:
Let's do this: let's move these two lines below the `improvedChildren` / `regressedChildren` definitions and pass those arrays into `attachLeaderToChildren`. Then you wouldn't need to do any selecting at all. Let's also move the `addEventListener` call into `attachLeaderToChildren`. That method would just the parent checkbox and children checkbox, and add a listener to the parent that toggles all the children.